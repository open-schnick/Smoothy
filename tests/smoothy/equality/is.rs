use crate::failing_assertion;
use smoothy::{assert_that, EqualityAssertion};

mod succeeds {
    use super::*;

    #[test]
    fn with_numbers() {
        assert_that(21u8).is(21);
    }

    #[test]
    fn with_strings() {
        assert_that(String::from("Hello World!")).is(String::from("Hello World!"));
    }

    #[test]
    fn with_string_slices() {
        assert_that("Hello World!").is("Hello World!");
    }
}

mod fails {
    use super::*;

    #[test]
    fn with_not_matching_values() {
        failing_assertion!({
            assert_that(21u8).is(12);
        });
    }
}
