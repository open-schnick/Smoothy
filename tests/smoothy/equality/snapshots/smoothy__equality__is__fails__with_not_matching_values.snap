---
source: tests/smoothy/equality/is.rs
expression: assertion_failed_output
---
assertion failed: `(actual == expected)`'
   actual: `"21"`
 expected: `"12"`

[1mDifferences[0m ([31m[2m-[0m[31mactual[0m|[32m[2m+[0m[32mexpected[0m):
[31m[1m[2m-[0m[31m21[0m
[32m[1m[2m+[0m[32m12[0m
